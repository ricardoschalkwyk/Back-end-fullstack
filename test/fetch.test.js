const supertest = require("supertest");

const app = require("../src/app");

let favorite;

describe("/search", () => {
  it("GET should return itunes results", async () => {
    const res = await supertest(app)
      .get("/api/search?term=rock")
      .expect("Content-Type", /json/)
      .expect(200);

    expect(res.body).toHaveProperty("result");
  });
});

describe("/favorites", () => {
  it("GET should return favorites", async () => {
    const res = await supertest(app)
      .get("/api/favorites")
      .expect("Content-Type", /json/)
      .expect(200);

    expect(res.body).toEqual([]);
    expect(res.body).toMatchSnapshot();
  });

  it("POST should create a new favorite", async () => {
    const res = await supertest(app)
      .post("/api/favorites/")
      .send({ artistName: "john", songName: "beans" })
      .set("Accept", "application/json")
      .expect("Content-Type", /json/)
      .expect(200);

    favorite = res.body;

    expect(res.body).toHaveProperty("id");
  });

  it("DELETE should delete single favorite", async () => {
    const res = await supertest(app)
      .delete(`/api/favorites/delete/${favorite.id}`)
      .expect("Content-Type", /json/)
      .expect(200);

    expect(res.body).toHaveProperty("deletedItem");
  });
});
