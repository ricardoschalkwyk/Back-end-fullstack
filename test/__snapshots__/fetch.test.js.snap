// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/favorites GET should return faorites 1`] = `Array []`;

exports[`/favorites GET should return favorites 1`] = `Array []`;
